ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java

```diff
@@ -268,4 +281,18 @@ private BlockHeader getChainHeadBlockHeader() {
 
     void onTransactionsAdded(Iterable<Transaction> transactions);
   }
+
+  private Wei minTransactionGasPrice(final Transaction transaction) {
```

Comment:
Style: I prefer to see returns at the same level, unless it is a guard block (https://wiki.c2.com/?GuardClause).  Unexpected fall throughs I've seen lead to bugs.

Both ways to address this will duplicate the frontier gas price calculation, but from a readability perspective I think it's fine as it makes what is going on explicit.

This could be re-written as two guard blocks (one checking for 1559 enablement in general, then the two variables calculation, then the check for basefee being present) returning frontier prices and then a final return for 1559 prices.

Reply:
I added a guard block, if EIP-1559 is not enabled in general, returns with frontier price.
Then i have a simple `if-else` block with returns at the same level. Actually i simplified because base fee is mandatory if the block is post EIP-1559, so i added an `orElseThrow` when retrieving the base fee.

Reply:
:+1: